cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/io/cdap/cdap/etl/spec/PipelineSpecGenerator.java

```
@@ -330,6 +328,41 @@ private ConfiguredStage configureStage(ETLStage stage, ValidatedPipeline validat
     return specBuilder;
   }
 
+  private Object getPlugin(String stageName, ETLPlugin etlPlugin, PluginSelector pluginSelector, String type,
+                           String pluginName, DefaultStageConfigurer stageConfigurer) {
+    Object plugin = null;
+    try {
+      plugin = pluginConfigurer.usePlugin(type, pluginName, stageName, etlPlugin.getPluginProperties(), pluginSelector);
+    } catch (IllegalArgumentException e) {
+      stageConfigurer.addFailure(e.getMessage(), String.format("Provide unique plugin id for plugin '%s'.", stageName));
```

IllegalArgument only happens with the plugin id is not unique? That seems to be a very general exception to use for a very specific type of error.

This should also never happen, the stageName is guaranteed to be unique by the pipeline code. If this happens, it's a bug in our application code.

---

Had an offline discussion with @albertshau . It makes sense to not catch pipeline app generated exceptions and let it propagate.

---

The implementations of `pluginConfigurer.usePlugin` method does not handle exceptions correctly.  Filed: https://issues.cask.co/browse/CDAP-15744 to fix it. 
For now, adding validation failure by capturing `Exception` instead of specific exceptions.